src/ddsrt/src/sync/posix/sync.c

```diff
@@ -158,25 +158,39 @@ ddsrt_rwlock_init (ddsrt_rwlock_t *rwlock)
 {
   assert(rwlock != NULL);
 
+#if __SunOS_5_6
```

Comment:
This would be better done by defining a macro that says whether `pthread_rwlock_...` is supported.

Reply:
Well, I agree with you comment, but it also wasn't the case before, so I'm a bit hesitant about pulling it into this changeset.

Reply:
If you don't mind, I'll leave this as is.